### ISSUE TITLE
hspy saving error: automatic chunk shape

### ISSUE THREAD

**Issue by magnunor** [hyperspy/hyperspy#2945]
When saving `hspy` files with specific sizes, the automatic chunking fails, giving an error.

#### Example:
```python
import numpy as np
import hyperspy.api as hs
s = hs.signals.Signal1D(np.zeros((3, 224, 256, 1024), dtype=np.uint8))
s.save("test.hspy", overwrite=True)
```

#### Gives the error 
```python
h5py/_hl/dataset.py:77, in make_new_dset(parent, shape, dtype, data, name, chunks, compression, shuffle, fletcher32, maxshape, compression_opts, fillvalue, scaleoffset, track_times, external, track_order, dcpl, allow_unknown_filter)
     72 if isinstance(chunks, tuple) and any(
     73     chunk > dim for dim, chunk in zip(tmp_shape, chunks) if dim is not None
     74 ):
     75     errmsg = "Chunk shape must not be greater than data shape in any dimension. "\
     76              "{} is not compatible with {}".format(chunks, shape)
---> 77     raise ValueError(errmsg)
     79 if isinstance(dtype, Datatype):
     80     # Named types are used as-is
     81     tid = dtype.id

ValueError: Chunk shape must not be greater than data shape in any dimension. (36, 9, 9, 1024) is not compatible with (3, 224, 256, 1024)
```

<details>
  <summary>Full error</summary>
  
```python
ValueError                                Traceback (most recent call last)
File 001_test_chunking_error_saving.py:4, in <module>
      2 import hyperspy.api as hs
      3 s = hs.signals.Signal1D(np.zeros((3, 224, 256, 1024), dtype=np.uint8))
----> 4 s.save("test.hspy", overwrite=True)

File hyperspy/signal.py:3004, in BaseSignal.save(self, filename, overwrite, extension, **kwds)
   3002     if extension is not None:
   3003         filename = filename.with_suffix(f".{extension}")
-> 3004 hyperspy.io.save(filename, self, overwrite=overwrite, **kwds)

File hyperspy/io.py:853, in save(filename, signal, overwrite, **kwds)
    851 signal = _add_file_load_save_metadata('save', signal, writer)
    852 if not isinstance(filename, MutableMapping):
--> 853     writer.file_writer(str(filename), signal, **kwds)
    854     _logger.info(f'{filename} was created')
    855     signal.tmp_parameters.set_item('folder', filename.parent)

File hyperspy/io_plugins/hspy.py:257, in file_writer(filename, signal, close_file, **kwds)
    255 try:
    256     writer = HyperspyWriter(f, signal, expg, **kwds)
--> 257     writer.write()
    258 except BaseException:
    259     raise

File hyperspy/io_plugins/_hierarchical.py:650, in HierarchicalWriter.write(self)
    649 def write(self):
--> 650     self.write_signal(self.signal,
    651                       self.group,
    652                       **self.kwds)

File hyperspy/io_plugins/_hierarchical.py:681, in HierarchicalWriter.write_signal(self, signal, group, write_dataset, chunks, **kwds)
    678 metadata_dict = signal.metadata.as_dictionary()
    680 if write_dataset:
--> 681     self.overwrite_dataset(
    682         group,
    683         signal.data,
    684         'data',
    685         signal_axes=signal.axes_manager.signal_indices_in_array,
    686         chunks=chunks,
    687         **kwds
    688         )
    690 if default_version < Version("1.2"):
    691     metadata_dict["_internal_parameters"] = \
    692         metadata_dict.pop("_HyperSpy")

File hyperspy/io_plugins/_hierarchical.py:629, in HierarchicalWriter.overwrite_dataset(cls, group, data, key, signal_axes, chunks, **kwds)
    621     these_kwds.update(dict(shape=data.shape,
    622                            dtype=data.dtype,
    623                            exact=True,
    624                            chunks=chunks,
    625                            ))
    627     # If chunks is True, the `chunks` attribute of `dset` below
    628     # contains the chunk shape guessed by h5py
--> 629     dset = group.require_dataset(key, **these_kwds)
    630     got_data = True
    631 except TypeError:
    632     # if the shape or dtype/etc do not match,
    633     # we delete the old one and create new in the next loop run

File h5py/_hl/group.py:229, in Group.require_dataset(self, name, shape, dtype, exact, **kwds)
    227 with phil:
    228     if not name in self:
--> 229         return self.create_dataset(name, *(shape, dtype), **kwds)
    231     if isinstance(shape, int):
    232         shape = (shape,)

File h5py/_hl/group.py:149, in Group.create_dataset(self, name, shape, dtype, data, **kwds)
    146         parent_path, name = name.rsplit(b'/', 1)
    147         group = self.require_group(parent_path)
--> 149 dsid = dataset.make_new_dset(group, shape, dtype, data, name, **kwds)
    150 dset = dataset.Dataset(dsid)
    151 return dset

File h5py/_hl/dataset.py:77, in make_new_dset(parent, shape, dtype, data, name, chunks, compression, shuffle, fletcher32, maxshape, compression_opts, fillvalue, scaleoffset, track_times, external, track_order, dcpl, allow_unknown_filter)
     72 if isinstance(chunks, tuple) and any(
     73     chunk > dim for dim, chunk in zip(tmp_shape, chunks) if dim is not None
     74 ):
     75     errmsg = "Chunk shape must not be greater than data shape in any dimension. "\
     76              "{} is not compatible with {}".format(chunks, shape)
---> 77     raise ValueError(errmsg)
     79 if isinstance(dtype, Datatype):
     80     # Named types are used as-is
     81     tid = dtype.id

ValueError: Chunk shape must not be greater than data shape in any dimension. (36, 9, 9, 1024) is not compatible with (3, 224, 256, 1024)
```
</details>

#### Python environement:
 - HyperSpy version: `1.7.0`
 - Python version: `3.9.12`

**Comment by francisco-dlp** [comment]
I think that this is a duplicate of #2940. There is a fix in #2943.

**Comment by magnunor** [comment]
Indeed!
